src/CMakeLists.txt

```diff
@@ -1,60 +1,76 @@
-include(MacroQbtCompilerSettings)
-qbt_set_compiler_options()
-
-include(QbtTargetSources)
-
+# find required packages for base and app
 find_package(Boost ${requiredBoostVersion} REQUIRED)
+#TODO: handle case where libtorrent was not built by cmake
 find_package(LibtorrentRasterbar ${requiredLibtorrentVersion} REQUIRED)
 find_package(OpenSSL ${requiredOpensslVersion} REQUIRED)
-
-if (Boost_VERSION VERSION_LESS 106000)
-    add_definitions(-DBOOST_NO_CXX11_RVALUE_REFERENCES)
-endif()
-
+find_package(ZLIB ${requiredZlibVersion} REQUIRED)
 find_package(Qt5 ${requiredQtVersion} REQUIRED COMPONENTS Core Network Xml LinguistTools)
-if (GUI)
-    find_package(Qt5Widgets ${requiredQtVersion} REQUIRED)
-    find_package(Qt5DBus ${requiredQtVersion})
+if (DBUS)
+    find_package(Qt5 ${requiredQtVersion} REQUIRED COMPONENTS DBus)
+    set_package_properties(Qt5DBus PROPERTIES
+        DESCRIPTION "Qt5 module for inter-process communication over the D-Bus protocol"
+        PURPOSE "Required by the DBUS feature"
+    )
 endif()
 
-set_package_properties(Qt5DBus PROPERTIES
-    DESCRIPTION "Qt5 module for inter-process communication over the D-Bus protocol"
-    PURPOSE "Enables communication with other system components (e.g. notification service) via D-Bus. "
-    TYPE RECOMMENDED
-)
+# automatically call Qt moc, rcc and uic as need for all targets by default
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTORCC ON)
+set(CMAKE_AUTOUIC ON)
 
-set(CMAKE_AUTOMOC True)
-list(APPEND CMAKE_AUTORCC_OPTIONS -compress 9 -threshold 5)
-if (APPLE)
-    # Workaround CMake bug (autogen does not pass required parameters to moc)
-    # Relevant issue: https://gitlab.kitware.com/cmake/cmake/issues/18041
-    list(APPEND CMAKE_AUTOMOC_MOC_OPTIONS -DQ_OS_MACOS -DQ_OS_DARWIN)
-endif()
+# compile options
+include(MacroQbtCompilerSettings)
+qbt_set_compiler_options()
 
-include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+# defines used in multiple targets, so they are added globally here
+add_compile_definitions(QBT_VERSION_MAJOR=${PROJECT_VERSION_MAJOR})
+add_compile_definitions(QBT_VERSION_MINOR=${PROJECT_VERSION_MINOR})
+add_compile_definitions(QBT_VERSION_BUGFIX=${PROJECT_VERSION_PATCH})
+add_compile_definitions(QBT_VERSION_BUILD=${PROJECT_VERSION_TWEAK})
```

Comment:
Why did these move from the top CMakeLists?

Reply:
Part of the effort to decouple anything related to adding compile definitions, adding compile options, include directories, etc. from the top-level file. This should make the transition to `target_*` commands in the future simpler.